runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5235,11 +5239,18 @@ class INTERPRETER_CLASS
 				UDATA result = preparePinnedVirtualThreadForUnmount(_currentThread, object, true);
 				VMStructHasBeenUpdated(REGISTER_ARGS);
 				if (J9_OBJECT_MONITOR_OOM != result) {
+					PORT_ACCESS_FROM_JAVAVM(_vm);
 					/* Handle the virtual thread Object.wait call.
 					 * VirtualThread.timeout is a private field used by both VM and JCL to temporarily hold
 					 * the value of expected wait/park time before a wake up task is scheduled using the value.
 					 */
-					J9VMJAVALANGVIRTUALTHREAD_SET_TIMEOUT(_currentThread, _currentThread->threadObject, millis + (nanos / 1000000));
+					I_64 timeout = millis + (nanos / 1000000);
```

Comment:
I don't think my [question](https://github.com/eclipse-openj9/openj9/pull/21590#discussion_r2033720193) about rounding up was answered.
I don't think the call `wait(0, 500000)` should not be treated the same as `wait(0, 0)`.

Reply:
I've update the comments to reflect the difference in unit between park and wait.
Also updated the nanos rounding to match RI https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/Object.java#L500-L502